src/Clock.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from "react";
+
+export default class Clock extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      time: new Date(),
+      hours: true,
+      ampm: false,
+      hour: 12,
+      minute: '00',
+      bn: 'am'
+      };
+  }
+
+  componentDidMount() {
+    this.timerId = setInterval(() => {
+      this.setState({
+        time: new Date()
+      });
+    }, 1000);
+  }
+
+  componentWillMount() {
+    clearInterval(this.timerId);
+  }
```

Comment:
Aside from `componentWillMount` being deprecated, this timeout seems completely redundant. You should receive the initial time from your props, and if you must then assign it with `setState` in `componentDidMount`.